### DISCUSSION TITLE
lsp command without invoking lsp--find-root-interactively

### DISCUSSION THREAD

**Discussion by vincentbernat** [emacs-lsp/lsp-mode#3433] [Q&A] [answered]
Hey!

Is there a way for the `lsp` command to not invoke `lsp--find-root-interactively`? I would like to use that as a hook for some modes without getting a prompt for unknown project (in this case, lsp should not start). Currently, when using `lsp-deferred` as a hook for `go-mode`, when I open a random Go file, I get the prompt and most of the time, it just eats my first keypress since I didn't expect it. I still would like to keep the prompt when invoking `lsp` interactively.

**Comment by yyoncho** [comment, marked as answer]
Can you be a bit more specific about what behavior you would like to achieve?

**Reply by vincentbernat** [reply]
When lsp is invoked through a hook and the project is unknown, I don't want to be prompted to import it. If I invoke lsp manually, I want to be prompted.

**Reply by yyoncho** [reply]
create your function which does the following:
```
(defun lsp-non-interactive ()
  (require 'lsp)
  (when (lsp-workspace-root)
    (lsp)))
```

and use it for the non-interactive version.

**Reply by vincentbernat** [reply]
Thanks!
